### ISSUE TITLE
Update and dynamic response header name

### ISSUE THREAD

**Issue by feteu** [arni-inaba/asgi-request-id#3]
Hi @arni-inaba 
Would you be open to receive a pull request which would add some standard python typing incl. the possibility to specify the response header name dynamically?

**Comment by arni-inaba** [comment]
@feteu sure! That sounds useful 🙂

**Comment by feteu** [comment]
@arni-inaba I've updated the outgoing header parameter as mentioned. Changing the typing will take a bit more time, as it may require restructuring the code to align with the latest Starlette/ASGI references. Let me know if you'd like me to proceed with that, though keep in mind it could break backward compatibility with Python 3.8. However, I can ensure compatibility down to Python 3.11. Let me know your thoughts...

**Comment by arni-inaba** [comment]
@feteu Adding typing would be great, we can just bump major when breaking Python 3.8.

**Comment by feteu** [comment]
Yes, I'll prepare something and open a new issue for that in parallel. I'll get back to you on that. I might target to support Python >=3.11, considering that 3.11 is already today only receiving security updates ([Python-EoL](https://devguide.python.org/versions/#supported-versions)).
